validator/delphi_validator/validate.py

```diff
@@ -244,9 +245,26 @@ def check_df_format(self, df_to_test, nameformat):
 
         self.increment_total_checks()
 
-    def check_bad_geo_id(self, df_to_test, nameformat, geo_type):
+    def check_bad_geo_id_value(self, df_to_test, geo_type):
         """
-        Check validity of geo type and values, according to regex pattern.
+        Check for bad geo_id values, by comparing to a list of known values (drawn from historical data)
+
+        Arguments:
+            - df_to_test: pandas dataframe of CSV source data containing the geo_id column to check
+            - geo_type: string from CSV name specifying geo type (state, county, msa, etc.) of data
+        """
+        file_path = join(self.validator_static_file_dir, geo_type + '_geo.csv')
+        valid_geo_df = pd.read_csv(file_path, dtype = {'geo_id': str})
+        valid_geos = valid_geo_df['geo_id'].values
+        unexpected_geos = [geo for geo in df_to_test['geo_id'] if geo not in valid_geos]
+        if len(unexpected_geos) > 0:
+            self.raised_errors.append(ValidationError(
+                ("check_bad_geo_id_value", filename),
+                unexpected_geos, "Unrecognized geo_ids (not in historical data)"))
+
```

Comment:
Increment check counter with `self.increment_total_checks()`.

Reply:
Does this still need to be fixed?

Reply:
Fixed in latest commit

Reply:
Yep, waiting on this and a few other changes!